### PR TITLE
Decouple the kubeseal CLI from the kubeseal library

### PR DESCRIPTION
**Pull request by alemorcuq**
Signed-off-by: Alejandro Moreno <amorenoc@vmware.com>

**Description of the change**
Decouple the kubeseal CLI specific code from its library.

Note: **The currently exposed `kubeseal` interface must be considered unstable, it will change going forward**.

**Benefits**
Makes the library easier to reuse.

**Possible drawbacks**
As with all refactors, bugs could be introduced. We will ensure however our test suite continues to pass.

### CODE REVIEW

**Comment by josvazg** [comment]
I believe it should now pass tests.

**Review by josvazg — DISMISSED** [review]
LGTM, but lets get some more approvals here

**Comment by josvazg** [comment]
BTW this will conflict once we merge #1029

**Comment by josvazg** [comment]
will check the conflicts locally instead

**Comment by josvazg** [comment]
Special mention: Recent removal of the `NamespaceFn` (see #1029)
That function was only needed for a single test (`TestRaw`) that required to customize the namespace used in such test. The way it was done was using a custom Interface `ClientConfig` that only container a couple of the methods from `clientcmd.ClienConfig` used by the kubeseal code here. It was ten trivial to use a wrapper object that re-implemented the `Namespace()` method for our test.

For the point of view if the current code, it might not even be needed to use the type wrapper trick above. But this change has tried to preserve that and leaves further refactor improvements for follow up PRs.

**Review by josvazg — DISMISSED** [review]

**Review by josvazg — APPROVED** [review]

**Review by alvneiayu — APPROVED** [review]
LGTM

**Comment by bakito** [comment]
@alemorcuq finally, thank you very much for this decoupling I was waiting for this for along time. 🥳

**Pull request merged**
